pydantic/networks.py

```diff
@@ -350,6 +397,98 @@ class PostgresDsn(AnyUrl):
     }
     user_required = True
 
+    __slots__ = AnyUrl.__slots__ + ('hosts',)
+
+    def __init__(self, *args, hosts: Optional[List['HostParts']] = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hosts = hosts
+
+    @classmethod
+    def validate_host_parts(cls, parts: 'HostParts') -> 'HostParts':
+        """
+        A method used to validate host parts of an URL.
+        """
+        port = parts['port']
+        if port is not None and int(port) > 65_535:
+            raise errors.UrlPortError()
+
+        return parts
+
+    @classmethod
+    def validate_multi_host(cls, hosts: List[str]) -> List['HostParts']:
+        hosts_parts: List['HostParts'] = []
+        for host in hosts:
+            hm = host_regex().match(host)
+            original_parts = cast('Parts', hm.groupdict())
+            host, tld, host_type, rebuild = cls.validate_host(original_parts)
+            host_parts = cast(
+                'HostParts',
+                {
+                    'host': host,
+                    'host_type': host_type,
+                    'tld': tld,
+                    'rebuild': rebuild,
```

Comment:
This is not correct. There's no point in using types if you're just going to ignore them.